pkg/types/types.go

```diff
@@ -234,39 +262,69 @@ func (s byStepTime) Less(i, j int) bool {
 	return s[i].StepTime < s[j].StepTime
 }
 
-func mergeMetrics(metrics []Metric) Metric {
+func getPointInconsistentValueCount(selectedValue float64, values []float64) int {
+	valuesCount := len(values)
+	if valuesCount == 0 {
+		// No value for the point
+		return 0
+	}
+
+	valuesToCount := make(map[float64]int)
+	for _, val := range values {
+		valuesToCount[val]++
+	}
+
+	return valuesCount - valuesToCount[selectedValue]
+}
+
+func mergeMetrics(metrics []Metric, consistencyCheck bool) (metric Metric, inconsistencies int) {
 	if len(metrics) == 0 {
-		return Metric{}
+		return Metric{}, 0
 	}
 
 	if len(metrics) == 1 {
-		return metrics[0]
+		m := metrics[0]
+		return m, 0
 	}
 
 	sort.Sort(byStepTime(metrics))
 	healed := 0
 
 	// metrics[0] has the highest resolution of metrics
-	metric := metrics[0]
+	metric = metrics[0]
 	for i := range metric.Values {
-		if !metric.IsAbsent[i] {
-			continue
+		var valuesForPoint []float64
```

Comment:
What if we just compare each replica to the original value instead of keeping all values in an array? It seems this might simplify the logic somewhat.

Reply:
Well, because I had kept an eye on the future implementation for handling inconsistencies (majority read, for instance), keeping the values and processing them later might be a good idea.

Reply:
My vote would be to not add complexity just in case it might be needed in the future. We could always add this later, once the actual need arises.

Reply:
please check @timtofan.